src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -154,7 +154,10 @@ public void Dispose()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
+                if (_solutionEvents != null)
```

Comment:
This could be a hint of some serious bug. `Dispose` called more than once? I think hiding this issue behind defensive statement is not the right thing. Let's file a separate issue and investigate.